lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -251,12 +251,12 @@ def compile(node, options)
     if node.environment && node.environment.static_catalogs? && options[:static_catalog] && options[:code_id]
       # Check for errors before compiling the catalog
       checksum_type = common_checksum_type(options[:checksum_type])
-      raise Puppet::Error, "Unable to find a common checksum type between agent '#{options[:checksum_type]}' and master '#{known_checksum_types}'." unless checksum_type
+      raise Puppet::Error, _("Unable to find a common checksum type between agent '%{agent_type}' and master '%{master_type}'.") % { agent_type: options[:checksum_type], master_type: known_checksum_types } unless checksum_type
     end
 
-    str = "Compiled %s for " % [checksum_type ? 'static catalog' : 'catalog']
+    str = _("Compiled %s for ") % [checksum_type ? _('static catalog') : _('catalog')]
```

Comment:
It was for sure this way already but it's really weird that we use `[` `]` as the markers here rather than `(` `)` as we do everywhere else. Any chance you could update to use the parens?